examples/stm32/nucleo-h723zg-make-baremetal-builtin/main.c

```diff
@@ -23,6 +23,16 @@ void mg_random(void *buf, size_t len) {  // Use on-board RNG
   }
 }
 
+#ifdef MQTT_DASHBOARD
+void hal_gpio_write(int pin, bool status) {  // For MQTT dashboard HAL
+  gpio_write((uint16_t) pin, status);
+}
+
+bool hal_gpio_read(int pin) {  // For MQTT dashboard HAL
+  return gpio_read((uint16_t) pin);
+}
+#endif
+
```

Comment:
This is not necessary if we do as in #2516 
I would also prefer to keep this away from the profiling stuff

Reply:
OK, there are additional benefits doing it your way, it is more flexible. There are some issues with "read" that this way solves.
I'll migrate #2516 stuff to this way.